ray-operator/controllers/ray/common/pod.go

```diff
@@ -243,32 +243,50 @@ func DefaultWorkerPodTemplate(instance rayv1.RayCluster, workerSpec rayv1.Worker
-				"&&", "bash", "-c", fmt.Sprintf("wget -T 2 -q -O- http://localhost:%d/%s | grep success",
-					utils.DefaultDashboardPort, utils.RayDashboardGCSHealthPath),
-			}
-			probe.Exec = &corev1.ExecAction{Command: cmd}
-		} else {
-			cmd := []string{
-				"bash", "-c", fmt.Sprintf("wget -T 2 -q -O- http://localhost:%d/%s | grep success",
-					utils.DefaultDashboardAgentListenPort, utils.RayAgentRayletHealthPath),
-			}
-			probe.Exec = &corev1.ExecAction{Command: cmd}
+func initLivenessAndReadinessProbe(rayContainer *corev1.Container, rayNodeType rayv1.RayNodeType, enableServeService bool) {
+	rayAgentRayletHealthCommand := fmt.Sprintf(utils.BaseWgetHealthCommand, utils.DefaultDashboardAgentListenPort, utils.RayAgentRayletHealthPath)
+	rayDashboardGCSHealthCommand := fmt.Sprintf(utils.BaseWgetHealthCommand, utils.DefaultDashboardPort, utils.RayDashboardGCSHealthPath)
+
+	// Generally, the liveness and readiness probes perform the same checks.
+	// For head node => Check GCS and Raylet status.
+	// For worker node => Check Raylet status.
+	commands := []string{}
+	if rayNodeType == rayv1.HeadNode {
+		commands = append(commands, rayAgentRayletHealthCommand, rayDashboardGCSHealthCommand)
+	} else {
+		commands = append(commands, rayAgentRayletHealthCommand)
+	}
+
+	if rayContainer.LivenessProbe == nil {
+		rayContainer.LivenessProbe = &corev1.Probe{
+			InitialDelaySeconds: utils.DefaultLivenessProbeInitialDelaySeconds,
+			TimeoutSeconds:      utils.DefaultLivenessProbeTimeoutSeconds,
+			PeriodSeconds:       utils.DefaultLivenessProbePeriodSeconds,
+			SuccessThreshold:    utils.DefaultLivenessProbeSuccessThreshold,
+			FailureThreshold:    utils.DefaultLivenessProbeFailureThreshold,
+		}
+		rayContainer.LivenessProbe.Exec = &corev1.ExecAction{Command: []string{"bash", "-c", strings.Join(commands, " && ")}}
+	}
+
+	if rayContainer.ReadinessProbe == nil {
+		rayContainer.ReadinessProbe = &corev1.Probe{
+			InitialDelaySeconds: utils.DefaultReadinessProbeInitialDelaySeconds,
+			TimeoutSeconds:      utils.DefaultReadinessProbeTimeoutSeconds,
+			PeriodSeconds:       utils.DefaultReadinessProbePeriodSeconds,
+			SuccessThreshold:    utils.DefaultReadinessProbeSuccessThreshold,
+			FailureThreshold:    utils.DefaultReadinessProbeFailureThreshold,
+		}
+		rayContainer.ReadinessProbe.Exec = &corev1.ExecAction{Command: []string{"bash", "-c", strings.Join(commands, " && ")}}
+
+		// For worker Pods serving traffic, we need to add an additional HTTP proxy health check for the readiness probe.
+		// Note: head Pod checks the HTTP proxy's health at every rayservice controller reconcile instaed of using readiness probe.
+		// See https://github.com/ray-project/kuberay/pull/1808 for reasons.
+		if enableServeService && rayNodeType == rayv1.WorkerNode {
+			rayContainer.ReadinessProbe.FailureThreshold = utils.ServeReadinessProbeFailureThreshold
+			rayServeProxyHealthCommand := fmt.Sprintf(utils.BaseWgetHealthCommand,
```

Comment:
Note: I executed the command `wget -T 2 -q -O- http://localhost:8000/-/healthz` in a RayService's worker Pod and received "success".